### PR TITLE
FIPS: add indicator for using EdDSA with a precomputed digest

### PR DESCRIPTION
**Pull request by paulidale**
NIST has mandates that a precalculated digest isn't permitted for EdDSA.
Instead the EdDSA operation needs to compute the digest.

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: doc/man7/EVP_SIGNATURE-ED25519.pod
Comment (slontis) [inline doc/man7/EVP_SIGNATURE-ED25519.pod:81]: back seems in the wrong place here

Comment (paulidale) [reply doc/man7/EVP_SIGNATURE-ED25519.pod:81]: It's matching elsewhere in the file.  The points are indented, the descriptive text isn't.

File: providers/implementations/signature/eddsa_sig.c
Comment (slontis) [inline providers/implementations/signature/eddsa_sig.c:600]: move this check up.. we should always do this first.

File: doc/man7/EVP_SIGNATURE-ED25519.pod
Comment (paulidale) [reply doc/man7/EVP_SIGNATURE-ED25519.pod:81]: The item is in the wrong place however.

File: providers/implementations/signature/eddsa_sig.c
Comment (paulidale) [reply providers/implementations/signature/eddsa_sig.c:600]: Why?

Comment (slontis) [reply providers/implementations/signature/eddsa_sig.c]: because if the params are all set in the same call it should be done first -before any other sets that use that value.

Comment (paulidale) [reply providers/implementations/signature/eddsa_sig.c]: Fair point but the set_ctx_params isn't doing any checking (that happens in the verify calls).
Anyway, I moved it already.

File: doc/man7/EVP_SIGNATURE-ED25519.pod
Comment (slontis) [inline doc/man7/EVP_SIGNATURE-ED25519.pod:41]: this is not right still.. I think listing the params and then listing them as items is not required.

Comment (paulidale) [reply doc/man7/EVP_SIGNATURE-ED25519.pod]: Better?

File: doc/man7/EVP_SIGNATURE-ED25519.pod
Comment (slontis) [inline doc/man7/EVP_SIGNATURE-ED25519.pod:46]: The following?

**Review by slontis — APPROVED** [review]

File: doc/man7/EVP_SIGNATURE-ED25519.pod
Comment (paulidale) [reply doc/man7/EVP_SIGNATURE-ED25519.pod:46]: Done.

**Comment by paulidale** [comment]
Ping @openssl/committers for second review.

**Comment by paulidale** [comment]
Rebased to avoid trivial conflict.  Re-ping for second review @openssl/committers

**Comment by paulidale** [comment]
Trivial conflict resolved (core_params array length update) during rebase to master.

**Comment by slontis** [comment]
I am not too concerned about this, considering there are multiple PR's that are going to hit this spot with a merge conflict.

**Comment by paulidale** [comment]
Re-ping @openssl/committers for second review please.

**Comment by paulidale** [comment]
Test failures aren't relevant, they are the ones from #25089.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the reviews.

**Pull request closed without merging**
